File: src/main/java/com/example/main/Runner.java
package com.example.main;

import com.example.entities.User;
import com.example.greeters.Greeter;
import com.example.repositories.UserDao;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;

@Component
public class Runner implements CommandLineRunner {

    @Autowired
    private UserDao userDao;

    @Override
    public void run(String... args) {
        var user1=new User("Neptune","neptune@example.com");

        userDao.save(user1);
    }


}

File: src/main/java/com/example/main/Runner.kt
package com.example.main

import com.example.entities.User
import org.springframework.boot.CommandLineRunner
import org.springframework.beans.factory.annotation.Autowired
import com.example.repositories.UserDao
import org.springframework.stereotype.Component

@Component
class Runner : CommandLineRunner {
    @Autowired
    private lateinit var userDao: UserDao

    override fun run(vararg args: String) {
        val user1 = User("Jupiter", "jupiter@example.com")
        val user = userDao.save(user1)
        val retrievedUserOpt = userDao.findById(user.id)
        if (retrievedUserOpt.isPresent) {
            println("Find by Id " + retrievedUserOpt.get())
        }
        userDao.findAll().forEach{u-> println(u)}
    }
}